packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -139,7 +144,8 @@ export function normalizeAttributeValue(
         // <input value={myValue} />
         // -> Valid identifier.
         return { value, escapedExpression: false };
-    } else if (!isEscaped && isPotentialExpression(value)) {
+    } else if (matchedBrackets && isPotentialExpression(value)) {
```

Comment:
Do we need to explicitly check for matched brackets when matched brackets are a prerequisite for potential expressions?